FBSnapshotTestCase/Categories/UIImage+Compare.m

```diff
@@ -129,4 +115,64 @@ - (BOOL)fb_compareWithImage:(UIImage *)image tolerance:(CGFloat)tolerance
     return imageEqual;
 }
 
+- (BOOL)comparePixelWithPixelTolerance:(CGFloat)pixelTolerance
+                        referencePixel:(FBComparePixel*)referencePixel
+                            imagePixel:(FBComparePixel*)imagePixel
+{
+    if (referencePixel->raw == imagePixel->raw) {
+        return YES;
+    } else if (pixelTolerance == 0) {
+        return NO;
+    }
+
+    CGFloat redPercentDiff = [self calculatePercentDifferenceForReferencePixelComponent:referencePixel->pixels.red
+                                                                    imagePixelComponent:imagePixel->pixels.red];
+    CGFloat greenPercentDiff = [self calculatePercentDifferenceForReferencePixelComponent:referencePixel->pixels.green
+                                                                      imagePixelComponent:imagePixel->pixels.green];
+    CGFloat bluePercentDiff = [self calculatePercentDifferenceForReferencePixelComponent:referencePixel->pixels.blue
+                                                                     imagePixelComponent:imagePixel->pixels.blue];
+    CGFloat alphaPercentDiff = [self calculatePercentDifferenceForReferencePixelComponent:referencePixel->pixels.alpha
+                                                                      imagePixelComponent:imagePixel->pixels.alpha];
+
+    return !(redPercentDiff > pixelTolerance ||
+             greenPercentDiff > pixelTolerance ||
+             bluePercentDiff > pixelTolerance ||
+             alphaPercentDiff > pixelTolerance);
+}
+
+- (CGFloat)calculatePercentDifferenceForReferencePixelComponent:(char)p1
+                                            imagePixelComponent:(char)p2
+{
+    int referencePixelComponent = (unsigned char)p1;
+    int imagePixelComponent = (unsigned char)p2;
+    int componentDifference = abs(referencePixelComponent - imagePixelComponent);
+    return (CGFloat)componentDifference/256;
+}
+
+- (BOOL)comparePixelWithPixelTolerance:(CGFloat)pixelTolerance
+                             tolerance:(CGFloat)tolerance
+                            pixelCount:(NSInteger)pixelCount
+                       referencePixels:(FBComparePixel*)referencePixel
+                           imagePixels:(FBComparePixel*)imagePixel
+{
+    NSInteger numDiffPixels = 0;
+    for (int n = 0; n < pixelCount; ++n) {
```

Comment:
If we're iterating over `0..pixelCount` where `pixelCount` is a `NSInteger`, then `n` should also be an `NSInteger`

Reply:
Done 👍